src/status_im/contexts/wallet/add_address_to_watch/view.cljs

```diff
@@ -71,84 +68,92 @@
-          :description-text (i18n/label :t/enter-eth)}]
-        [:f> address-input
-         {:input-value    input-value
-          :validate       validate
-          :validation-msg validation-msg
-          :clear-input    clear-input}]
-        (when @validation-msg
-          [quo/info-message
-           {:accessibility-label :error-message
-            :size                :default
-            :icon                :i/info
-            :type                :error
-            :style               style/info-message}
-           @validation-msg])
-        [activity-indicator]]])))
+      (let [activity-state    (rf/sub [:wallet/watch-address-activity-state])
+            validated-address (rf/sub [:wallet/watch-address-validated-address])]
+        [rn/view {:style {:flex 1}}
+         [floating-button-page/view
+          {:header [quo/page-nav
+                    {:type      :no-title
+                     :icon-name :i/close
+                     :on-press  (fn []
+                                  (rf/dispatch [:wallet/clean-scanned-address])
+                                  (rf/dispatch [:wallet/clear-address-activity])
+                                  (rf/dispatch [:navigate-back]))}]
+           :footer [quo/button
+                    {:customization-color customization-color
+                     :disabled?           (or (string/blank? @input-value)
+                                              (some? (validate @input-value))
+                                              (= activity-state :invalid-ens)
+                                              (= activity-state :scanning)
+                                              (not validated-address))
+                     :on-press            (fn []
+                                            (rf/dispatch [:navigate-to
+                                                          :confirm-address-to-watch
+                                                          {:address validated-address}])
+                                            (clear-input))
+                     :container-style     {:z-index 2}}
+                    (i18n/label :t/continue)]}
+          [quo/page-top
+           {:container-style  style/header-container
+            :title            (i18n/label :t/add-address)
+            :description      :text
+            :description-text (i18n/label :t/enter-eth)}]
+          [:f> address-input
+           {:input-value    input-value
+            :validate       validate
+            :validation-msg validation-msg
+            :clear-input    clear-input}]
+          (if @validation-msg
+            [quo/info-message
+             {:accessibility-label :error-message
+              :size                :default
+              :icon                :i/info
+              :type                :error
+              :style               style/info-message}
+             @validation-msg]
+            [activity-indicator activity-state])]]))))
```

Comment:
This `if` was a `when`, while using `when`, two messages could be displayed under certain circumstances, because we have two components and the back-end response comes some time later.

The message is the following:
![image](https://github.com/status-im/status-mobile/assets/90291778/16c4dafa-71d6-4c35-a3b5-a22fa63994e0)

I think we should only have a single component, having two makes the code more error-prone. A bigger refactor on this component might be needed, but I didn't do it here because I wanted to keep the diff small.